drivers/network/dd/e1000/hardware.c

```diff
@@ -0,0 +1,822 @@
+/*
+ * PROJECT:     ReactOS Intel PRO/1000 Driver
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Hardware specific functions
+ * COPYRIGHT:   2018 Mark Jansen (mark.jansen@reactos.org)
+ *              2019 Victor Perevertkin (victor.perevertkin@reactos.org)
+ */
+
+#include "nic.h"
+
+#include <debug.h>
+
+
+static USHORT SupportedDevices[] =
+{
+    0x1001,     // Intel 82543GC Fiber
+    0x1004,     // Intel 82543GC Copper
+    0x100e,     // Intel 82540EM
+    0x100f,     // Intel 82545EM
```

Comment:
You forgot to add
- 0x1011 / 82545EM (Intel PRO/1000 MF Server LX or SX)
- 0x1026 / 82545GM (Intel PRO/1000 MT Server)
- 0x1027 / 82545GM (Intel PRO/1000 MF Server LX or SX)

Reply:
Has someone actually tested these cards with this driver?
As much as we'd all like to have our drivers work for more cards, it's been stated multiple times already both by members of the ReactOS dev team, and by multiple of the specific developers that worked on this driver that they don't want to add the IDs for cards that haven't been tested yet.

Once someone has tested that these cards work properly with this driver, I'm certain the devs would love to add them to the list.

Reply:
If you had checked the pdf that Iv provided you would have realized that the 82545EM controller match 3 cards which were already mentioned in CORE-14675. These cards have 2 ids each. The MF Server ones share the same ids and were missing. The second id of the MT Server was missing.

With your logic we should remove 0x100f which match the MT Server as well.
Anyway if anything we could add even more ids. For now I am just requesting the one listed in the original issue.